src/tiered.js

```diff
@@ -67,14 +93,23 @@ class TieredDatastore extends Adapter {
     await Promise.all(this.stores.map(store => store.close()))
   }
 
+  /**
+   * @returns {Batch}
+   */
   batch () {
     const batches = this.stores.map(store => store.batch())
```

Comment:
Inline type annotations below seem to suggest that `batches` aren’t inferred here for some reason, it might be better to fix that or at least add `/** @type {Batch[]} */` to make below annotations unnecessary